mbox/mbox.c

```diff
@@ -1866,6 +1866,7 @@ struct MxOps MxMboxOps = {
   .path_canon       = mbox_path_canon,
   .path_pretty      = mbox_path_pretty,
   .path_parent      = mbox_path_parent,
+  .path_is_empty    = mutt_file_check_empty,
```

Comment:
This will sound inefficient, but can you create an `mbox_path_is_empty()` function to call.

The APIs have an arrays of function pointers.
When debugging them, it's useful if the functions have unique names,
so in `mx_path_is_empty()` the pointer `ops->path_is_empty` is mbox-specific.

Reply:
want me to duplicate the function body? Or would you rather have library function called from both?

rewrote here... replying via mail breaks the threading apparently

Reply:
@flatcap so... this is an issue over almost all the MxOps structs... not only in that field.

it's pretty prominent in mh vs maildir... they both share quite a lot of code and have random methods of the other one in their ops.
In fact, the only backends that consistently have that are compress (not that this is hard...) / imap and nntp

Do you really want me to wrap those all in function calls?

Reply:
MMDF will be especially painfull... it shares 90% of the code with mbox

Reply:
Hang on!
I just wanted a single function wrapper around the `mutt_file_check_empty()` call.
Nothing else.

So that all the functions in MxMboxOps and MxMmdfOps have names beginning either `mbox_` or `mmdf_`.

Reply:
yes. However most other backends have also methods from other backends embedded. The obly places that have this consistently is imap, nntp and  compress.

If your issue is debugging you still have cases where say the notmuch backend does stuff but the function pointer is called maildirsdomething, which was your original complaint no?

Reply:
> which was your original complaint no?

Yes, but one thing at a time :-)
I'll get to the other backends eventually.